oprf/oprf.go

```diff
@@ -160,94 +236,125 @@ func mustWrite(h io.Writer, bytes []byte) {
 	}
 }
 
-func (s *suite) computeComposites(
-	k group.Scalar,
-	B group.Element,
-	Cs []group.Element,
-	Ds []group.Element,
-) (group.Element, group.Element, error) {
-	Bm, err := B.MarshalBinaryCompress()
-	if err != nil {
-		return nil, nil, err
-	}
+var (
+	ErrInvalidSuite       = errors.New("invalid suite")
+	ErrInvalidMode        = errors.New("invalid mode")
+	ErrDeserialization    = errors.New("deserialization failed")
+	ErrDeriveKeyPairError = errors.New("key pair derivation failed")
+	ErrInvalidInput       = errors.New("invalid input")
+	ErrInvalidInfo        = errors.New("invalid info")
+	ErrInvalidProof       = errors.New("proof verification failed")
+	ErrInverseZero        = errors.New("inverting a zero value")
+	ErrNoKey              = errors.New("must provide a key")
+)
 
-	lenBuf := []byte{0, 0}
-	H := s.New()
+type (
+	blind     = group.Scalar
+	blinded   = group.Element
+	evaluated = group.Element
+)
 
-	binary.BigEndian.PutUint16(lenBuf, uint16(len(Bm)))
-	mustWrite(H, lenBuf)
-	mustWrite(H, Bm)
+// FinalizeData encapsulates data needed for Finalize step.
+type FinalizeData struct {
+	inputs  [][]byte
+	blinds  []blind
+	evalReq *EvaluationRequest
+}
 
-	dst := s.getDST(seedDST)
-	binary.BigEndian.PutUint16(lenBuf, uint16(len(dst)))
-	mustWrite(H, lenBuf)
-	mustWrite(H, dst)
+// EvaluationRequest contains the blinded elements to be evaluated by the Server.
+type EvaluationRequest struct {
+	g        group.Group
+	elements []blinded
```

Comment:
I think this should be exported so that applications can encode these as necessary. Right now, users of this API _have_ to use the `MarshalBinary` interface below, which might not be what every application wants or needs.

Reply:
internals of protocol messages are public.
Similar to what is requested in #318

Reply:
Excellent!